<a href="https://colab.research.google.com/github/svkarthik86/Fundamentals-of-statistics-and-probability/blob/main/Week2_Assignment_Answers_ipynb_txt.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### WEEK-2: Assignment  - Probability Distribution

### 1.Calculate Probabilities Using a Binomial Distribution

Q1.1. Pavan kumar makes 65% of his free-throw attempts. If he shoots 15 free throws, what is the probability that he makes exactly 10?

P(X=10)   where X=k=10,no of trials n=15,Sucess Probabibilty p=0.65 

In [2]:
from scipy.stats import binom
binom.pmf(k=10,n=15,p=0.65)


0.21233868348803567

Q1.2.Shivam flips a fair coin 6 times. What is the probability that the coin lands on heads 3 times or fewer?

P(X<=3), n=6,p=0.5

In [3]:
from scipy.stats import binom
binom.cdf(k=3,n=6,p=0.5)

0.6562499999999999

Q1.3.It is known that 65% of individuals support a certain law. If 10 individuals are randomly selected, what is the probability that between 3 and 6 of them support the law?

P(3<X<6)   n=10, p=.65

In [4]:
from scipy.stats import binom
binom.cdf(k=6,n=10,p=0.65)-binom.cdf(k=3,n=10,p=0.65)

0.4601487031476564

### 2.Calculate Probabilities Using a Poisson Distribution

Q2.1. A store sells 4 apples per day on average. What is the probability that they will sell 6 apples on a given day? 

P(X=6) ,mu=4

In [9]:
from scipy.stats import poisson
poisson.pmf(k=6,mu=4)


0.10419563456702102

Q2.2. A certain store sells seven footballs per day on average. What is the probability that this store sells four or less footballs in a given day?

P(X<=4) , mu=7

In [10]:
from scipy.stats import poisson
poisson.cdf(k=4,mu=7)

0.17299160788207146

Q2.3. A certain store sells 15 cans of tuna per day on average. What is the probability that this store sells more than 20 cans of tuna in a given day?

P(X>20)=1-P(X<=20),  where mu=15

In [11]:
from scipy.stats import poisson
1-poisson.cdf(k=20,mu=15)

0.08297091003146029

### 3.Calculate Probabilities Using a Uniform Distribution

Q3.1. Suppose a bus shows up at a bus stop every 20 minutes. If you arrive at the bus stop, what is the probability that the bus will show up in 8 minutes or less?

In [12]:
from scipy.stats import uniform
#calculate uniform probability
uniform.cdf(x=8, loc=0, scale=20) - uniform.cdf(x=0, loc=0, scale=20)

0.4

Q3.2.The weight of a certain species of frog is uniformly distributed between 15 and 25 grams. If you randomly select a frog, what is the probability that the frog weighs between 17 and 19 grams?

In [13]:
from scipy.stats import uniform
#calculate uniform probability
uniform.cdf(x=19, loc=15, scale=10) - uniform.cdf(x=17, loc=15, scale=10)

0.2

Q3.3.The length of an NBA game is uniformly distributed between 120 and 170 minutes. What is the probability that a randomly selected NBA game lasts more than 150 minutes?

In [14]:
from scipy.stats import uniform
#calculate uniform probability 
uniform.cdf(x=170, loc=120, scale=50) - uniform.cdf(x=150, loc=120, scale=50)


0.4

### 4.Calculate Probabilities Using a Normal Distribution

Q4.1.A radar unit is used to measure speeds of cars on a motorway. The speeds are normally distributed with a mean of 90 km/hr and a standard deviation of 10 km/hr. What is the probability that a car picked at random is travelling at more than 100 km/hr?

x has μ = 90 and σ = 10. We have to find the probability that x is higher than 100 or P(x > 100)

norm.pdf(x, loc, scale)
The location (loc) keyword specifies the mean. The scale (scale) keyword specifies the standard deviation.

In [26]:
from scipy.stats import norm
1-norm.cdf(x=100,loc=90,scale=10)

0.15865525393145707

Q4.2.For a certain type of computers, the length of time bewteen charges of the battery is normally distributed with a mean of 50 hours and a standard deviation of 15 hours. John owns one of these computers and wants to know the probability that the length of time will be between 50 and 70 hours.

Let x be the random variable that represents the length of time. It has a mean of 50 and a standard deviation of 15. We have to find the probability that x is between 50 and 70 or P( 50< x < 70)

In [27]:
from scipy.stats import norm
norm.cdf(x=70,loc=50,scale=15)-norm.cdf(x=50,loc=50,scale=15)

0.4087887802741321


Q4.3.Entry to a certain University is determined by a national test. The scores on this test are normally distributed with a mean of 500 and a standard deviation of 100. Tom wants to be admitted to this university and he knows that he must score better than at least 70% of the students who took the test. Tom takes the test and scores 585. Will he be admitted to this university?

Let x be the random variable that represents the scores. x is normally ditsributed with a mean of 500 and a standard deviation of 100. The total area under the normal curve represents the total number of students who took the test. If we multiply the values of the areas under the curve by 100, we obtain percentages.
For x = 585 , z = (585 - 500) / 100 = 0.85

The proportion P of students who scored below 585 is given by P(X<=585)


In [29]:
from scipy.stats import norm
norm.cdf(x=585,loc=500,scale=100)

0.8023374568773076

**Tom scored better than 80.23% of the students who took the test and he will be admitted to this University.**